prs/WeatherData.cs

```diff
@@ -0,0 +1,35 @@
+namespace Sample {
+    /**
+     * A class which updates displays with various atmospheric data.
+     */
+    public class WeatherData {
+
+        private float temperature;
+        private float humidity;
+        private float pressure;
+        private Location location;
+
+        public WeatherData(float temperature, float humidity, float pressure, Location location) {
+            this.temperature = temperature;
+            this.humidity = humidity;
+            this.pressure = pressure;
+            this.location = location;
+        }
+
+        public void updateMeasurements(float? temperature, float? humidity, float? pressure) {
```

Comment:
One of my favorite features in C# is [properties](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties). Rather than having one method for updating each of the measurements, can we expose property setters?